alchemy/src/scope.ts

```diff
@@ -537,6 +550,49 @@ export class Scope {
     return promise;
   }
 
+  /**
+   * Install global process exit handlers to ensure cleanup
+   */
+  private installExitHandlers() {
+    if (this.parent) return;
+    onExit((code) => {
+      if (this.cleanups.length === 0) return;
+      this.logger.log(kleur.gray("Exiting..."));
+      Promise.allSettled(this.cleanups.map((cleanup) => cleanup())).then(() => {
+        console.log("Exiting with code:", code); // this never gets logged, thanks Miniflare
+        process.exit(code);
+      });
+      return true;
+    });
+  }
+
+  /**
+   * Register a cleanup function that will be called when the scope is finalized.
+   * The provided function should create resources and return an object with a cleanup function.
+   * This should only be called on the root scope to ensure proper cleanup.
+   */
+  public async spawn<T extends { cleanup: () => Promise<void> }>(
```

Comment:
That's not what this function does

Reply:
It seems overly complicated. Why not just onCleanup?

It's also not clear why the user provides a function that we immediately call

Reply:
Fair enough